packages/oasis/package.py

```diff
@@ -0,0 +1,93 @@
+    of coupling information between numerical codes representing
+    different components of the climate system."""
+
+    homepage = "https://portal.enes.org/oasis"
+    git      = 'https://gitlab.com/cerfacs/oasis3-mct.git'
+    maintainers = ['pheidippides']
+
+    version('master', branch='OASIS3-MCT_4.0')
+    version('dev-build', branch='OASIS3-MCT_4.0')
+
+    depends_on('mpi', type=('build', 'link', 'run'))
+    depends_on('netcdf-fortran', type=('build','link','run'))
+
+    build_directory = 'util/make_dir'
+
+    makefile_file = 'TopMakefileOasis3'
+
+    # Relative path where the built libraries are stored (corresponds
+    # to the absolute path called ARCHDIR in the Makefile)
+    rel_ARCHDIR='spack-build'
+
+    
+    def setup_build_environment(self, env):
+
+        CHAN = 'MPI1'
+        env.set('CHAN', CHAN)
+        env.set('F90', self.spec['mpi'].mpifc)
+        env.set('f90', self.spec['mpi'].mpifc)
+        env.set('F', self.spec['mpi'].mpifc)
+        env.set('f', self.spec['mpi'].mpifc)
+        env.set('MAKE', 'gmake')
+
+        LIBBUILD = os.path.join('../..', self.rel_ARCHDIR, 'build/lib')
+        INCPSMILE = '-I{LIBBUILD}/psmile.{CHAN} -I{LIBBUILD}/mct -I{LIBBUILD}/scrip'.format(LIBBUILD=LIBBUILD, CHAN=CHAN)
+
+        
+        CPPDEF = '-Duse_comm_{CHAN} -D__VERBOSE -DTREAT_OVERLAY -D__NO_16BYTE_REALS'.format(CHAN=CHAN)
+        env.set('CPPDEF', CPPDEF)
+
+        FFLAGS = '-O2 {INCPSMILE} {CPPDEF}'.format(CPPDEF=CPPDEF, INCPSMILE=INCPSMILE)
+        env.set('F90FLAGS', FFLAGS)
+        env.set('f90FLAGS', FFLAGS)
+        env.set('FFLAGS', FFLAGS)
+        env.set('fFLAGS', FFLAGS)
+        env.set('CCFLAGS', FFLAGS)
+
+
+    def edit(self, spec, prefix):
+
+        COUPLE = os.getcwd()
+        ARCHDIR = os.path.join(COUPLE, self.rel_ARCHDIR)
+        with working_dir(self.build_directory):
+            makefile = FileFilter(self.makefile_file)
+            makefile.filter('include make.inc', 'export COUPLE = {}\nexport ARCHDIR = {}'.format(COUPLE, ARCHDIR))
+            makefile.filter('\$\(modifmakefile\)\s\;\s' , '')
+
+
+    def patch(self):
+
+        # Remove old directives for Fujitsu comilers. Already fixed in MCT [1] but not merged in OASIS yet
+        # [1] https://github.com/MCSclimate/MCT/commit/dcb4fa4527bbc51729fb67fbc2e0179bfcb4baa2
+        with working_dir('lib/mct/mct'):
+            m_AttrVect = FileFilter('m_AttrVect.F90')
+            m_AttrVect.filter('\s*\!DIR\$ COLLAPSE', '')
+        
+
+    def build(self, spec, prefix):
+        
+        with working_dir(self.build_directory):
+            make('-f', self.makefile_file)
+
+
+    def install(self, spec, prefix):
+        
+        with working_dir(os.path.join(self.rel_ARCHDIR, 'lib')):
+            os.symlink('libmct.a', 'libmct_oasis.a')
+            os.symlink('libmpeu.a', 'libmpeu_oasis.a')
+            
+        copytree(self.rel_ARCHDIR, prefix, dirs_exist_ok=True, symlinks=True)
+        rmtree(self.rel_ARCHDIR)
```

Comment:
This line is obsolete, since Spack will remove the build-stage anyway after installation.

Reply:
I don't think spack knows this is the directory where the Makefile stores the compiled libraries. It would of course be deleted in case of a `spack install` command but not in case of `spack dev-build`. Still this is a strategic question. You might want to keep that directory after a `spack dev-build`.